.pre-commit-config.yaml

```diff
@@ -0,0 +1,27 @@
+# CHECK for new hook to add (JS, etc.): https://pre-commit.com/hooks.html
```

Comment:
Can you back this out into a separate PR? I don't like adding build deps as a "surprise!"

Reply:
Sorry, I do not understand what building deps it adds. Can you elaborate please?

Reply:
Are the pre-commit checks purely a developer client-side check?   So they have no effect on the CI build chain (or external ones)?

Reply:
Exactly, these are purely dev side checks and are not mandatory to work. They may affect the CI if you integrate the pre-commit checks in the CI (which is not the case).

Reply:
Hi,

As a follower of this work and an intensive user of git hooks through pre-commit, I just want to mention that it is possible to plug https://pre-commit.ci/ in order to run it automatically on PR.

Reply:
Thanks @Guts I think we will soon add theses checks in the github CI